### PR TITLE
Reserve Space for AMP Ads &  Apply UX Specifications

### LINKED ISSUE #7380: Reserve space to prevent content shifting when Ads are rendered on AMP
**Is your feature request related to a problem? Please describe.**
We use a combination of the values from the *config service* and *GeoIP* to determine whether or not display to show display Ads. We currently check the config service on the server but we will still be checking this on the client because the value of this may be outdated as AMP pages may have been in the AMP cache for long. These checks on the client will cause content reflow when the Ads are loaded in. 
This follows from this [investigation doc](https://paper.dropbox.com/doc/Investigation-Content-Shifting-for-Amp-Ads--A4sCkGZIdXzBvIBJz55uQ6SQAg-Rki4cZvd0cN56zFHuyB85)

**Describe the solution you'd like**

- We first check if we should render an ad on the server, using the value from the config service. If this value says we shouldn’t render an ad, then we won’t. No space reserved, no content shifting.
- If the config service says we should render an ad slot, then we do, and reserve the space for the ad on the server render.
- Inside this reserved space, on the client we call the config service again, to make sure the value hasn’t recently changed (as pages could be in the AMP cache for a long time). If this returns false now, we don’t render the ad into this reserved space.

So, we only render a space with no ad in if the config service has been switched from true to false in the time between the server render of the page, and the user accessing the page via the AMP cache. This will then cause the AMP cache to request a new version of the page, updating it to a version without ads for subsequent users.

Also considering geoip lookup:

- Inside the reserved space, we can check to see if the user is within the UK. If they are, we will not render an ad. This means that users will have the reserved empty space at the top of the page. This was deemed acceptable for initial release (although should be confirmed by Jon/Jim/Gail).

- Further investigation - can we collapse the space reserved for an ad once it is no longer in the viewport once we have the geoip response? Using different container types potentially?

### UX Specifications
**Min Screen Width**
If screen width is below minimum, Ads should not be shown
Leaderboard: 320px
MPU: 300px

**Content Shifting**
We are not allowing any content shifting on AMP Ads

### Ad Unit Size

Leaderboard | Mobile: [320px - 1007px] | Desktop: [1008px+]
-- | -- | --
Ad Units Served | 320px x 50px300px x 50px | 970px x 250px 728px x 90px

MPU | All Breakpoints
-- | --
Ad Units Served | 300px x 250px

### Rendering the Ad Slot

We will always render a space the width of the screen. We will then vary the Ad Slot height depending on breakpoint.

**Padding**
The additional space we need to render for the grey border around the largest Ad Unit for each Breakpoint.

Leaderboard - Group 1, 2, and 3: `8px` of padding
Leaderboard-  Custom Leaderboard Breakpoint: `16px` of padding
MPU - All Breakpoints: `8px` of padding

**‘Advertisement’ Link**
Above each Ad Unit should be a link with the word ‘Advertisement’, language varied on service. The size of the 'Advertisement’ link above the Ad Unit is `25px`. 

- Font size: 11 
- Colour:  `#3F3F42`
- Should be regular - not bold

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by itimirichard**
Resolves #7380

**Overall change:**
_Apply UX specs to reserve space on the server for AMP Ads._

**Code changes:**

   - _Added `ampLeaderboardStyles` and `ampMpuStyles` to adSlotStyles.js._
   - _Added flex, flex-direction, and justify-content for vertical centering._
   - _Ensure a client fetch of the toggles config occurs on render._
   - _Updated snapshots._
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Review by sareh — APPROVED** [review]
The implementation looks good. 
When loading an AMP front page locally ([docs here](https://github.com/bbc/simorgh/blob/latest/docs/Rendering-Ads.md)) and visiting an AMP front page with ads set to true on the remote toggle, http://localhost:7080/mundo.amp?site=test&ads-debug=true, I see the placeholders at mobile and desktop breakpoints:

Screenshot MPU 360px
Screenshot Leaderboard 1000px

Screenshot Leaderboard 360px
Screenshot MPU 1000px

I throttled the connection to a slow 3g connection to check that the placeholders are present on the server render.

I did the same for a service that has ads disabled (http://localhost:7080/thai.amp?site=test&ads-debug=true) with a throttled slow 3g connection and saw that there were no placeholders for the AMP ads here. 
Screenshot AMP Service without ads - no placeholder

File: src/app/containers/Ad/Amp/index.jsx [hunk 1]
```diff
@@ -112,29 +105,31 @@ const AmpAd = ({ slotType }) => {
...
-          {AMP_ACCESS_FETCH(service)}
-        </Helmet>
-
-        <StyledAd>
-          <StyledLink
-            href={LABEL_LINK}
-            script={script}
-            service={service}
-            dir={dir}
+      <Helmet>
```
Comment (DenisHdz) [inline src/app/containers/Ad/Amp/index.jsx:105]: Comparing this container with the `Canonical` one, I'm wondering if we can move the `Helmet` bit above the `AdSection` for consistency.

File: src/app/containers/Ad/Amp/index.jsx [hunk 2]
```diff
@@ -29,27 +28,25 @@ const Section = props => <section {...props} />;
 // setting display: none ensures ad requests within this component are not made.
```
Comment (DenisHdz) [inline src/app/containers/Ad/Amp/index.jsx]: The above comments will need to be moved to line 44 now.

File: src/app/containers/Ad/Amp/index.jsx (hunk 1 above)
Comment (itimirichard) [reply src/app/containers/Ad/Amp/index.jsx:105]: Thanks @DenisHdz . This has been done to match the structure of the Canonical

**Review by DenisHdz — APPROVED** [review]
LGTM, great work 👍

**Comment by PriyaKR** [comment]
Testing:

By blocking the request to config on the network tab was not able to reproduce the condition for the secnario where in the reserved space would be seen.Also pairing with @itimirichard found that hardcoding the `amp-access` value to false the behaviour was intermittent(ads were shown or ads were not shown with no reserved space being shown).

**Comment by itimirichard** [comment]
> Testing:
> 
> By blocking the request to config on the network tab was not able to reproduce the condition for the secnario where in the reserved space would be seen.Also pairing with @itimirichard found that hardcoding the `amp-access` value to false the behaviour was intermittent(ads were shown or ads were not shown with no reserved space being shown).

@Priya, the issue with the amp-access value has been fixed now. This was happening because styled-components was stripping the amp attributes added to that div on render. This has now been adrressed and when I set the amp-access value to false, the ad and ad link are not shown, and the space is reserved.

**Comment by PriyaKR** [comment]
Local Testing:
Looks good to me.
User accessing the afrique amp page from uk for which the ads are enabled in the config -> Reserved blank space is rendered with no ads
User accessing the afrique amp page from uk for which the ads are disabled in the config -> No ads and no reserved space is rendered
User accessing the afrique amp page from non uk for which ads are enabled when the server fetch is done and when the client fetch happens the ads are disabled(locally this scenario was stimulated by hard coding the `amp-access` value to false -> No ads and reserved blank space is rendered.On live this scenario would happen when when amp page is served from google cache which has a page with ads enabled and when the client side rendering happens if the ads are disabled then reserved blank space will be rendered instead of ads.

reserved_blank_space

**Pull request merged**